### PR TITLE
Find the right IAM and ACM Certificate on senza create

### LINKED ISSUE #219: Find the right IAM and ACM Certificate on senza create
senza create use currently a _stupid_ way to select the IAM certificate. They replace all points (.) in the domain with minus (-) and try to use this.

Better: Read the issued domain names form all IAM and ACM certificate and match with with most matched and newest certificate.

### PR DESCRIPTION
**Pull request by jmcs**
Fixes #219.

### CODE REVIEW

**Comment by tuxlife** [comment]
@jmcs I can't see, that you check the domains names in IAM certificates and i can't see the support for multiple renewed domains

File: tests/test_manaus/test_iam.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
+from datetime import datetime, timezone
+from unittest.mock import MagicMock
+
+from senza.manaus.iam import IAM, IAMServerCertificate
+
+IAM_CERT1 = {'CertificateBody': 'body',
+             'CertificateChain': 'chain',
+             'ServerCertificateMetadata': {
+                 'Arn': 'arn:aws:iam::0000:server-certificate/senza-example-com',
```
Comment (tuxlife) [inline tests/test_manaus/test_iam.py]: You can't reuse a used Name for a new Certificate.

You must add a prefix like the upload date

**Comment by jmcs** [comment]
@tuxlife to check to domains of the IAM certificates I would need to add  pyOpenSSL as a dependency to parse the certificate, which needs to compile an extension module. What do you mean by "multiple renewed domains"?

**Comment by tuxlife** [comment]
I mean with "multiple renewed domains" a multidomain certificate which renewed with additional Domainnames. This looks like:
```
» aws iam list-server-certificates
{
    "ServerCertificateMetadataList": [
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/team-example-org",
            "Expiration": "2022-05-24T00:00:00Z",
            "Path": "/",
            "ServerCertificateName": "team-example-org",
            "ServerCertificateId": "...",
            "UploadDate": "2015-05-28T09:17:53Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/team-example-org-20160429",
            "Expiration": "2023-04-28T00:00:00Z",
            "Path": "/",
            "ServerCertificateName": "team-example-org-20160429",
            "ServerCertificateId": "...",
            "UploadDate": "2016-04-29T13:23:37Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/multidomain-lounge-shop-comodo-20160518",
            "Expiration": "2016-11-16T23:59:59Z",
            "Path": "/",
            "ServerCertificateName": "multidomain-lounge-shop-comodo-20160518",
            "ServerCertificateId": "...",
            "UploadDate": "2016-05-18T11:49:27Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/multidomain-example-org-comodo-20160415",
            "Expiration": "2018-09-13T23:59:59Z",
            "Path": "/",
            "ServerCertificateName": "multidomain-example-org-comodo-20160415",
            "ServerCertificateId": "...",
            "UploadDate": "2016-04-15T10:23:01Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/multidomain-example-org-comodo-20160425",
            "Expiration": "2018-09-13T23:59:59Z",
            "Path": "/",
            "ServerCertificateName": "multidomain-example-org-comodo-20160425",
            "ServerCertificateId": "...",
            "UploadDate": "2016-04-25T17:06:05Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/multidomain-example-org-comodo-20160502",
            "Expiration": "2018-09-13T23:59:59Z",
            "Path": "/",
            "ServerCertificateName": "multidomain-example-org-comodo-20160502",
            "ServerCertificateId": "...",
            "UploadDate": "2016-05-02T16:10:38Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/multidomain-example-org-comodo-20160509",
            "Expiration": "2018-09-13T23:59:59Z",
            "Path": "/",
            "ServerCertificateName": "multidomain-example-org-comodo-20160509",
            "ServerCertificateId": "...",
            "UploadDate": "2016-05-09T15:35:46Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/multidomain-example-org-comodo-20160606",
            "Expiration": "2018-09-13T23:59:59Z",
            "Path": "/",
            "ServerCertificateName": "multidomain-example-org-comodo-20160606",
            "ServerCertificateId": "...",
            "UploadDate": "2016-06-06T13:08:38Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/multi-example-net-comodo",
            "Expiration": "2018-09-13T23:59:59Z",
            "Path": "/",
            "ServerCertificateName": "multi-example-net-comodo",
            "ServerCertificateId": "...",
            "UploadDate": "2015-09-14T15:28:35Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/multi-example-net-comodo-20151022",
            "Expiration": "2018-09-13T23:59:59Z",
            "Path": "/",
            "ServerCertificateName": "multi-example-net-comodo-20151022",
            "ServerCertificateId": "...",
            "UploadDate": "2015-10-22T17:30:59Z"
        },
        {
            "Arn": "arn:aws:iam::123123123:server-certificate/foobar-example-org-comodo-20160606",
            "Expiration": "2020-03-17T23:59:59Z",
            "Path": "/",
            "ServerCertificateName": "foobar-example-org-comodo-20160606",
            "ServerCertificateId": "...",
            "UploadDate": "2016-06-06T13:00:45Z"
        }
    ]
}

```

In this example we update the **multidomain-example-org-comodo** multiple times.

### Summary of our offline discussion
- IAM Certificates: Support Names with Date suffix
  - The name _multidomain-example-org-comodo_ will find the `multidomain-example-org-comodo` and `multidomain-example-org-comodo-\d{8}`
  - The last one from the sorted list is used from Senza
  - If no name given, then use the default domain (. replaced by -) as name and support the date suffix
- ACM Certificate: Right Certificate is find by the domain entrys and not only the Name (primary domain)

File: tests/test_manaus/test_iam.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                     'Expiration': datetime(2015, 6, 29, 0, 0),
+                     'Path': '/',
+                     'ServerCertificateId': 'CERTIFICATEID',
+                     'ServerCertificateName': 'senza-example-com-20150701',
+                     'UploadDate': datetime(2015, 7, 1, 16, 0, 40)}}
+
+IAM_CERT2 = {'CertificateBody': 'body',
+             'CertificateChain': 'chain',
+             'ServerCertificateMetadata': {
+                 'Arn': 'arn:aws:iam::0001:server-certificate/senza-example-com',
```
Comment (tuxlife) [inline tests/test_manaus/test_iam.py]: This must be `'Arn': 'arn:aws:iam::0001:server-certificate/senza-example-com-20150702',`

File: tests/test_manaus/test_iam.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+from senza.manaus.iam import IAM, IAMServerCertificate
+
+IAM_CERT1 = {'CertificateBody': 'body',
+             'CertificateChain': 'chain',
+             'ServerCertificateMetadata': {
+                 'Arn': 'arn:aws:iam::0000:server-certificate/senza-example-com',
+                 'Expiration': datetime(2022, 6, 29, 0, 0),
+                 'Path': '/',
+                 'ServerCertificateId': 'CERTIFICATEID',
+                 'ServerCertificateName': 'senza-example-com-20150701',
```
Comment (tuxlife) [inline tests/test_manaus/test_iam.py]: This must be `'ServerCertificateName': 'senza-example-com',`

**Comment by tuxlife** [comment]
The Testcase is wrong:

1. ARN and Certificate Name is not the same
2. wrong certificate are selected. If you have ´senza-example-com´ and ´senza-example-com-20150701´ then you must use sort the list and use the last one (**senza-example-com-20150701**)

**Comment by jmcs** [comment]
@tuxlife I fixed the ARN. On `/components/elastic_load_balancer.py` the sort order is reversed to account for the default sort order (which is ordered according to the normal conventions for date sorting).

**Comment by tuxlife** [comment]
:+1:

**Comment by rafaelcaricio** [comment]
👍

**Pull request merged**